### PR TITLE
add SpecialFunctions to [deps]

### PR DESCRIPTION
**Pull request by JeffreySarnoff**
imported in src/specialfunctions.jl

### CODE REVIEW

**Comment by JeffreySarnoff** [comment]
@simonbyrne pls merge if ok with you -- I am pulling Quadmath into DoubleFloats for `sin`,`cos`,`tan` with large arguments (takes about the same time, more accurate).  It helps for Quadmath to add cleanly.

**Comment by simonbyrne** [comment]
We currently use Requires.jl for SpecialFunctions.jl: does that not work?

I'm not opposed to adding it (since SpecialFunctions.jl is already used by a lot of packages): if we do, we should remove Requires.jl.

**Comment by JeffreySarnoff** [comment]
without the Dep, there is a warning generated

**Comment by simonbyrne** [comment]
Ah, I think the issue is we need to `import .SpecialFunctions`.

**Comment by JeffreySarnoff** [comment]
❓ 

```
  Testing Quadmath
 Resolving package versions...
ERROR: LoadError: InitError: LoadError: LoadError: UndefVarError: @testset not defined
Stacktrace:
in expression starting at /home/travis/build/JuliaMath/Quadmath.jl/test/specfun.jl:1
```

File: src/Quadmath.jl [hunk 1]
```diff
@@ -107,9 +106,7 @@ reinterpret(::Type{Float128}, x::Int128) =
     reinterpret(Float128, reinterpret(UInt128, x))
 
 function __init__()
```
Comment (simonbyrne) [inline src/Quadmath.jl]: remove `__init__` block, and move the `include` statement to the bottom (next to the printf one)

**Pull request merged**